src/main/java/com/bytecoders/pharmaid/service/SharingPermissionService.java

```
@@ -0,0 +1,163 @@
+package com.bytecoders.pharmaid.service;
+
+import com.bytecoders.pharmaid.repository.SharingPermissionRepository;
+import com.bytecoders.pharmaid.repository.UserRepository;
+import com.bytecoders.pharmaid.repository.model.SharingPermission;
+import com.bytecoders.pharmaid.repository.model.SharingPermissionStatus;
+import com.bytecoders.pharmaid.repository.model.SharingRequest;
+import com.bytecoders.pharmaid.repository.model.User;
+import java.util.Date;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+/**
+ * Service class providing helper functions to manage sharing requests and permissions.
+ * Handles creation, acceptance, denial, and revocation of sharing requests between users.
+ */
+@Slf4j
+@Service
+public class SharingPermissionService {
+
+  @Autowired
+  private SharingPermissionRepository sharingPermissionRepository;
+
+  @Autowired
+  private UserRepository userRepository;
+
+  /**
+   * Creates a new sharing request between users.
+   *
+   * @param ownerId The ID of the user who owns the records.
+   * @param request The sharing request details.
+   * @return String The ID of the created sharing permission.
+   * @throws RuntimeException if either the owner or requester is not found.
+   */
+  public String createSharingRequest(String ownerId, SharingRequest request) {
+
+    User owner =
+        userRepository.findById(ownerId).orElseThrow(() -> new RuntimeException("Owner not found"));
```

you need to throw a specific exception. This should throw a 404